File: ex-13/ex-13_python/Vetor.py
class Vetor:
    # Construtoras
    def __init__(self, x=0, y=0):
        self.dx = x
        self.dy = y

    # Set's
    def setDx(self, x):
        self.dx = x

    def setDy(self, y):
        self.dy = y

    # Get's
    def getDx(self):
        return self.dx

    def getDy(self):
        return self.dy

    # Soma dois vetores
    def soma(self, v):
        return Vetor(v.dx + self.dx, v.dy + self.dy)

    # Multiplica um vetor por uma constante
    def multiplica(self, i):
        return Vetor(self.dx * i, self.dy * i)

File: ex-13/ex-13_java/Vetor.java
public class Vetor
{
    private int dx, dy;

    //Construtoras
    public Vetor(int x, int y)
    {
        dx = x;
        dy = y;
    }

    public Vetor()
    {
        dx = 0;
        dy = 0;
    }

    //Set's
    public void setDx(int x)
    {
        dx = x;
    }

    public void setDy(int y)
    {
        dy = y;
    }

    //Get's
    public int getDx()
    {
        return dx;
    }

    public int getDy()
    {
        return dy;
    }

    //Soma dois vetores
    public Vetor soma(Vetor v)
    {
        return new Vetor(v.dx + dx, v.dy + dy);
    }

    //Multiplica um vetor por uma constante
    public Vetor multiplica(int i)
    {
        return new Vetor(dx * i, dy * i);
    }
}
